### DISCUSSION TITLE
!![Error using the GRASS-section and GRASS-script libraries in the Python environment]

### DISCUSSION THREAD

**Discussion by elenaalf** [OSGeo/grass#3326] [Q&A]
Hello, 
I need to use GRASS from Python. In particular, I have installed the following packages with the specified versions in a Python environment with Python 3.9 interpreter:
grass 0.1.2
grass-script 0.0.4
grass-session 0.5
However, every time I try to import them, it gives me the [ModuleNotFoundError] error. I don't understand why. On my computer, I have installed GRASS GIS version 7.8 and QGIS 3.28.11.

Thank you so much!
Elena :)

**Comment by marisn** [comment]
Hello,
If you installed GRASS with `pip install grass`, then you got a totally different software: Graph Regularized Annotation via Semi-Supervised learning
Same goes for `grass-script` (A Python module that is basically a programming language) 
Can't comment on the current state of `grass-session`.

Besides, do not use GRASS 7 unless you really have to. Use 8.3 instead.
I usually just run my Python scripts from a GRASS session (start GRASS and then call `python myscript.py`) as it takes care of setting up environment. In the past I used `grass -e` approach to automate execution of python scripts from a work session.

**Reply by elenaalf** [reply]
Thank you so much!!

**Reply by neteler** [reply]
In addition, see also:

https://grasswiki.osgeo.org/wiki/Working_with_GRASS_without_starting_it_explicitly#Python:_GRASS_GIS_8_with_existing_location

**Reply by elenaalf** [reply]
Hi. I followed this tutorial, but this error keeps occurring: import grass.script as gs
ModuleNotFoundError: No module named 'grass'. Do you have any suggestions? I am sure that in the virtual environment the grass section was installed

**Reply by marisn** [reply]
GRASS files do not go into virtualenv. Easiest way is to just start normal GRASS session and then run python from within a session. Then `import grass.script as gs` will work without any extra steps.

**Reply by nilason** [reply]
You could also try setting PYTHONPATH:

```sh
export PYTHONPATH=$(grass --config path)/etc/python 
```

**Reply by nilason** [reply]
or better:
```sh
export PYTHONPATH=$(grass --config python_path)
```
